sapi/cli/php_cli.c

```diff
@@ -1137,6 +1140,12 @@ static int do_cli(int argc, char **argv) /* {{{ */
 	if (translated_path) {
 		free(translated_path);
 	}
+	if (behavior == PHP_MODE_LINT && argc > php_optind && strcmp(argv[php_optind],"--")) {
+		script_file = NULL;
+		request_started = 0;
```

Comment:
Move `request_started` and `translated_path` after `php_request_shutdown((void *) 0);` and `free(translated_path);` respectively?